tasks/nunjucks.js

```diff
@@ -6,65 +6,159 @@
+
+                        // Skip to next source file — nothing we can do with non-existing file
+                        return reject('Some source files were not found.')
+                    }
+
+                    // Construct absolute path to file for Nunjucks
+                    let filepath = path.join(process.cwd(), src)
+
+                    let data = {}
+                    // Work with data only if there is any data
+                    if (options.data) {
+                        // Clone data
+                        for (let i in options.data) {
+                            if (options.data.hasOwnProperty(i)) {
+                                data[i] = options.data[i]
+                            }
+                        }
+
+                        // Preprocess data
+                        if (typeof options.preprocessData === 'function') {
+                            data = options.preprocessData.call(file, data)
+                        }
+                    }
+
+                    // Asynchronously render templates with configurated Nunjucks environment
+                    // and write to destination
+                    env.render(filepath, data, (error, result) => {
+                        // Catch errors, warn
+                        if (error) {
+                            grunt.log.error(error)
+                            grunt.fail.warn('Failed to compile one of the source files.')
+                            grunt.log.writeln()
+
+                            // Prevent writing of failed to compile file, skip to next file
+                            return reject('Failed to compile some source files.')
+                        }
+
+                        // Write rendered template to destination
+                        grunt.file.write(filedest, result)
+
+                        // Debug process
+                        grunt.verbose.ok(`File ${chalk.cyan(filedest)} created.`)
+                        grunt.verbose.writeln()
+
+                        countCompiled++
+                    })
+                })
+            })
+
+            // Finish Promise
+            resolve()
+        })
+
+        // Print any errors from rejects
+        .catch(error => {
+            if (error) {
+                grunt.log.writeln()
+                grunt.log.error(error)
+                grunt.log.writeln()
             }
-            completeTask();
-        });
-    });
-};
+        })
+
+        // Log number of processed templates
+        .then(success => {
+            // Log number of processed templates
+            let logType            = (countCompiled === totalFiles) ? 'ok' : 'error'
+            let countCompiledColor = (logType === 'ok') ? 'green' : 'red'
+            grunt.log[logType](`${chalk[countCompiledColor](countCompiled)}/${chalk.cyan(totalFiles)} ${grunt.util.pluralize(totalFiles, 'file/files')} compiled.`)
+        })
+
+        // Finish async task
+        completeTask()
```

Comment:
It should be in `then()`, but for now it't located outside because of described at the beginning behavior — if code will hit `.catch` during first loops, it will run `.then` immediately after it. If `.then` will hold `completeTask()` inside, task will be finished immediately, without rendering all other files